itest/rfq_test.go

```diff
@@ -234,8 +234,19 @@ func testRfqAssetSellHtlcIntercept(t *harnessTest) {
 	ctxt, cancel := context.WithTimeout(ctxb, defaultWaitTimeout)
 	defer cancel()
 
-	// TODO(ffranr): Add an asset buy offer to Bob's tapd node. This will
-	//  allow Alice to sell the newly minted asset to Bob.
+	// Upsert an asset buy offer to Bob's tapd node. This will allow Bob to
+	// buy the newly minted asset from Alice.
+	_, err := ts.BobTapd.AddAssetBuyOffer(
```

Comment:
Is this diff here enough to also test the expected behavior of the itest?

Or I assume not all pieces are here yet for the e2e flow?

IIUC here we only test that the RFQ buy offer was placed, but not used in any way

Reply:
With the updates in this PR, the integration test fails unless we introduce an asset buy offer to Bob's node, as demonstrated here. This is necessary because, in the absence of the buy offer, Alice's subsequent request would be instantly declined by Bob's node, owing to the unavailability of an appropriate buy offer.

The goal of this itest is to test the entire payment flow facilitated by the RFQ service. We are not explicitly testing that the buy offer is used.